pkg/xds/sync/egress_proxy_builder.go

```diff
@@ -94,24 +95,21 @@ func (p *EgressProxyBuilder) Build(
 		}
 
 		for _, es := range externalServices {
-			policies := core_xds.PluginOriginatedPolicies{}
-			for name, plugin := range core_plugins.Plugins().PolicyPlugins() {
-				egressPlugin, ok := plugin.(core_plugins.EgressPolicyPlugin)
-				if !ok {
-					continue
-				}
-				res, err := egressPlugin.EgressMatchedPolicies(es, meshCtx.Resources)
-				if err != nil {
-					return nil, errors.Wrapf(err, "could not apply policy plugin %s", name)
-				}
-				if res.Type == "" {
-					return nil, errors.Errorf("matched policy didn't set type for policy plugin %s", name)
-				}
-				policies[res.Type] = res
+			policies, err := matchExternalPolicies(es.Spec.GetTags(), meshCtx.Resources)
+			if err != nil {
+				return nil, err
 			}
 			meshResources.Dynamic[es.Spec.GetService()] = policies
 		}
 
+		for serviceName := range meshResources.EndpointMap {
+			policies, err := matchExternalPolicies(map[string]string{mesh_proto.ServiceTag: serviceName}, meshCtx.Resources)
+			if err != nil {
+				return nil, err
+			}
+			meshResources.Dynamic[serviceName] = policies
```

Comment:
Won't it affect other plugins? @lobkovilya

Reply:
It shouldn't affect other plugins, we are only adding possible services on which we can apply policies

Reply:
It seems like it only affects the situations when we have an internal service with the same `kuma.io/service` tag as external?